core/src/main/java/org/opensearch/sql/calcite/CalciteRexNodeVisitor.java

```diff
@@ -195,7 +197,9 @@ public RexNode visitCompare(Compare node, CalcitePlanContext context) {
         transferCompareForDateRelated(leftCandidate, context, whetherCompareByTime);
     final RexNode right =
         transferCompareForDateRelated(rightCandidate, context, whetherCompareByTime);
-    return context.relBuilder.call(op, left, right);
+    return PPLFuncImpTable.INSTANCE
+        .resolve(node.getOperator(), List.of(left, right))
+        .apply(context.rexBuilder, left, right);
```

Comment:
Method name confusion: how about change to following? 
```
.apply(node.getOperator(), List.of(left, right))
.resolve(context.rexBuilder, left, right);
```

`apply()` in `FunctionImpX` normally should return an instance of `FunctionImpX`